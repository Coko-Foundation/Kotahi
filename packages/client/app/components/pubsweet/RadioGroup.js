/* eslint-disable react/destructuring-assignment */

import React from 'react'

import Radio from './Radio'
import Flexbox from './Flexbox'

class RadioGroup extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      value: props.value,
    }
  }

  handleChange = event => {
    const { value } = event.target
    this.setState({ value })
    this.props.onChange(value)
  }

  render() {
    const { className, disabled, inline, name, options, required } = this.props
    const { value } = this.state

    return (
      <Flexbox column={!inline}>
        {options.map(option => (
          <Radio
            checked={option.value === value}
            className={className}
            color={option.color}
            disabled={disabled || option.disabled}
            inline={inline}
            key={option.value}
            label={option.label}
            name={name}
            onChange={this.handleChange}
            required={required}
            value={option.value}
          />
        ))}
      </Flexbox>
    )
  }
}

export default RadioGroup
