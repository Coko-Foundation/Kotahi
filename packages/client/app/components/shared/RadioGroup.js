import React from 'react'
import styled from 'styled-components'
import { Radio } from './Radio'

const Flexbox = styled.div`
  display: flex;
  flex-direction: ${props => (props.column ? 'column' : 'row')};
  justify-content: ${props => (props.center ? 'center' : 'left')};
`

// eslint-disable-next-line import/prefer-default-export
export const RadioGroup = ({
  options,
  value: radioValue,
  'data-testid': dataTestid,
  className,
  disabled,
  inline,
  required,
  onChange,
}) => {
  const handleChange = event => {
    const { value } = event.target
    onChange(value)
  }

  return (
    <Flexbox column={!inline} data-testid={dataTestid}>
      {options.map(option => (
        <Radio
          checked={option.value === radioValue}
          className={className}
          color={option.color}
          disabled={disabled || option.disabled}
          inline={inline}
          key={option.value}
          label={option.label}
          name={option.name}
          onChange={handleChange}
          required={required}
          value={option.value}
        />
      ))}
    </Flexbox>
  )
}
