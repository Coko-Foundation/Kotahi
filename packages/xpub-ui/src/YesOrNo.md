A group of radio buttons that provides just two options: "Yes" or "No"

```js
initialState = { foo: undefined };

<YesOrNo
  name="foo" 
  value={state.foo} 
  onChange={event => setState({ foo: event.target.value })}/>
```
