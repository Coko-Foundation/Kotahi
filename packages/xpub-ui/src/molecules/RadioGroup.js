import React from 'react'
import Radio from '../atoms/Radio'

const RadioGroup = ({ name, value, options, required }) => (
  <div>
    {options.map(option => (
      <Radio
        key={option.value}
        name={name}
        required={required}
        value={option.value}
        label={option.label}
        checked={option.value === value}/>
    ))}
  </div>
)

export default RadioGroup
