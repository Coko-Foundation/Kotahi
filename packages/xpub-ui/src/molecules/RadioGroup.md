A group of radio buttons.

```js
initialState = { foo: undefined };

const options = [
  {
    value: 'one',
    label: 'One'
  },
  {
    value: 'two',
    label: 'Two'
  },
  {
    value: 'three',
    label: 'Three'
  }
];

<RadioGroup 
  options={options} 
  name="foo"
  value={state.foo} 
  handleChange={foo => setState({ foo })}/>
```
