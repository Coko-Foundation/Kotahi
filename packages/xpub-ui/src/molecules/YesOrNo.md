A group of radio buttons that provides just two options: "Yes" or "No"

```js
<YesOrNo
  name="yesorno" 
  onChange={event => console.log(event.target.value)}/>
```

If a value is set, one option is selected.

```js
<YesOrNo
  name="yesorno-value" 
  value="yes"
  onChange={event => console.log(event.target.value)}/>
```

