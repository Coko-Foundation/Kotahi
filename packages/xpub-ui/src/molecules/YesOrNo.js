import React from 'react'
import RadioGroup from './RadioGroup'
import classes from './YesOrNo.local.css'

const options = [
  {
    value: 'yes',
    label: 'Yes'
  },
  {
    value: 'no',
    label: 'No'
  }
]

const YesOrNo = ({ name, value }) => (
  <RadioGroup
    className={classes.root}
    name={name}
    options={options}
    value={value}/>
)

export default YesOrNo
