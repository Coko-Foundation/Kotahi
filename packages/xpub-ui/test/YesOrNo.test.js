import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import YesOrNo from '../src/molecules/YesOrNo'
import RadioGroup from '../src/molecules/RadioGroup'

const handleChange = () => { return null }

const props = {
  handleChange,
  name: 'TestName',
  value: 'Maybe'
}

const wrapper = shallow(<YesOrNo {...props} />)
const radio = wrapper.find(RadioGroup)

describe('Yes or No', () => {
  test('Snapshot', () => {
    const tree = renderer.create(
      <YesOrNo {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Renders a RadioGroup', () => {
    expect(wrapper.is('RadioGroup')).toBeTruthy()
    expect(radio).toHaveLength(1)
  })

  test('Passes the correct options', () => {
    const options = radio.props().options
    expect(options).toHaveLength(2)

    expect(options[0].value).toEqual('yes')
    expect(options[0].label).toEqual('Yes')

    expect(options[1].value).toEqual('no')
    expect(options[1].label).toEqual('No')
  })

  test('Passes down the correct name', () => {
    const name = radio.props().name
    expect(name).toEqual(props.name)
  })

  test('Passes down the correct value', () => {
    const value = radio.props().value
    expect(value).toEqual(props.value)
  })

  test('Passes down the correct handle change function', () => {
    const handle = radio.props().handleChange
    expect(handle).toEqual(props.handleChange)
  })
})
