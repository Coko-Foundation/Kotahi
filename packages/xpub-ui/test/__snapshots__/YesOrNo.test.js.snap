// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Yes or No Snapshot 1`] = `
<div>
  <label
    className={undefined}
  >
    <input
      checked={false}
      className={undefined}
      name="TestName"
      required={undefined}
      type="radio"
      value="yes"
    />
    Yes
  </label>
  <label
    className={undefined}
  >
    <input
      checked={false}
      className={undefined}
      name="TestName"
      required={undefined}
      type="radio"
      value="no"
    />
    No
  </label>
</div>
`;
